jetty-webapp/src/test/java/org/eclipse/jetty/webapp/ConfigurationsTest.java

```diff
@@ -110,6 +112,131 @@ public void testConfigurations()
+                ConfigHarry.class.getName(),
+                ConfigAdditionalHarry.class.getName(),
+                ConfigFoo.class.getName(),
+                ReplacementDick.class.getName()
+        );
+
+        Configurations configs = new Configurations(
+                ConfigBar.class.getName(),
+                ConfigZ.class.getName(),
+                ConfigY.class.getName(),
+                ConfigX.class.getName(),
+                ConfigTom.class.getName(),
+                ReplacementDick.class.getName(),
+                ConfigHarry.class.getName(),
+                ConfigAdditionalHarry.class.getName(),
+                ConfigFoo.class.getName()
+        );
+        
+        //ReplacementDick is already in the list and thus ConfigDick should not be added
+        configs.add(ConfigDick.class.getName());
+        configs.sort();
+        
+        assertThat(configs.stream().map(c->c.getClass().getName()).collect(toList()),
+                contains(
+                        ConfigFoo.class.getName(),
+                        ConfigBar.class.getName(),
+                        ConfigX.class.getName(),
+                        ConfigY.class.getName(),
+                        ConfigZ.class.getName(),
+                        ConfigTom.class.getName(),
+                        ReplacementDick.class.getName(),
+                        ConfigHarry.class.getName(),
+                        ConfigAdditionalHarry.class.getName()
+                        ));
+        
+        assertThat(configs.stream().map(c->c.getClass().getName()).collect(toList()),
+                   not(contains(
+                           ConfigDick.class.getName()
+                           )));
+    }
+    
+    @Test
+    public void testTransitiveReplacements () throws Exception
+    {
+        Configurations.cleanKnown();
+        Configurations.setKnown(
+                ConfigBar.class.getName(),
+                ConfigZ.class.getName(),
+                ConfigY.class.getName(),
+                ConfigX.class.getName(),
+                ConfigTom.class.getName(),
+                ConfigDick.class.getName(),
+                ConfigHarry.class.getName(),
+                ConfigAdditionalHarry.class.getName(),
+                ConfigFoo.class.getName(),
+                ReplacementDick.class.getName(),
+                AnotherReplacementDick.class.getName()
+        );
+
+        Configurations configs = new Configurations(
+                ConfigBar.class.getName(),
+                ConfigZ.class.getName(),
+                ConfigY.class.getName(),
+                ConfigX.class.getName(),
+                ConfigTom.class.getName(),
+                AnotherReplacementDick.class.getName(),
+                ConfigDick.class.getName(),
+                ConfigHarry.class.getName(),
+                ConfigAdditionalHarry.class.getName(),
+                ConfigFoo.class.getName()
+        );
+        
+        configs.add(ConfigDick.class.getName());
```

Comment:
`ConfigDick` is already in the configs, as it `AnotherReplacementDick`
You need to add `ReplacementDick` and then test that `ConfigDick` is removed,  `ReplacementDick` is not added and `AnotherReplacementDick` is left in the set

Reply:
I think it would have been clearer if you called the classes: ConfigDickA, ConfigDickB and ConfigDickC

Reply:
Well, I was working with the names that you'd already created in the original test.